stan/math/rev/mat/fun/positive_ordered_constrain.hpp

```diff
@@ -0,0 +1,80 @@
+#ifndef STAN_MATH_REV_MAT_FUN_POSITIVE_ORDERED_CONSTRAIN_HPP
+#define STAN_MATH_REV_MAT_FUN_POSITIVE_ORDERED_CONSTRAIN_HPP
+
+#include <stan/math/prim/arr/err/check_nonzero_size.hpp>
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/math/rev/mat/fun/adj_jac_apply.hpp>
+#include <vector>
+
+namespace stan {
+namespace math {
+
+namespace {
+class positive_ordered_constrain_op {
+  int N_;
+  double* exp_x_;
+
+ public:
+  positive_ordered_constrain_op() : N_(0) {}
```

Comment:
[question]
When does the nullary constructor ever get called?  If never, just get rid of it.

Same question for the other ones.